Flat/src/GeneralTree.cc

```diff
@@ -34,6 +34,11 @@ GeneralTree::GeneralTree() {
       }
     }
   }
+  
+  for (unsigned iShift=0; iShift<17; iShift++) {
```

Comment:
Can we avoid hardcoding 17?

Reply:
maybe

Reply:
Answer is no because the enums are not sequential integers. But we can hardcode 17 in the header